com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/util/BuildInputDataCollector.java

```diff
@@ -0,0 +1,137 @@
+package com.dynamo.bob.util;
+
+import com.dynamo.bob.archive.EngineVersion;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.FileWriter;
+import java.nio.file.Files;
+import java.security.MessageDigest;
+
+public class BuildInputDataCollector {
+    private static final String DATA_FILE_NAME = "build_input_data.json";
+
+    private static String[] bobArguments;
+    private static Map<String, File> dependencies;
+
+    public static void setArgs(String[] args) {
+        bobArguments = args;
+    }
+
+    public static void setDependencies(Map<String, File> libFiles) {
+        dependencies = libFiles;
+    }
+
+    public static void saveDataAsJson(String rootDirectory, File bundleOutputDirectory) {
+        TimeProfiler.start("BuildInputDataCollector");
+        Map<String, Object> data = new HashMap<>();
+        data.put("BobArguments", bobArguments);
+
+        Map<String, String> engineVersion = new HashMap<>();
+        engineVersion.put("sha1", EngineVersion.sha1);
+        engineVersion.put("version", EngineVersion.version);
+        engineVersion.put("timestamp", EngineVersion.timestamp);
```

Comment:
Tbh, this is actually only guaranteed to be the Bob version + timestamp.

In many cases, ppl use the `--defold-sdk` to select engine version.
While it's then part of the arguments, it may be nice if we add it explicitly to a json-field for easier extraction?

Reply:
Fixed!